module-bsp/board/rt1051/bsp/magnetometer/magnetometer.cpp

```diff
@@ -118,234 +122,280 @@ namespace bsp
+            lastMeasurement            = currentMeasurement;
 
-            // any configuration must be proceeded in active state
-            setActive(als31300::PWR_REG_SLEEP_MODE::active);
+            return positionChanged;
+        }
+    } // namespace
 
-            // GET WRITE ACCESS
-            if (!i2cWrite(als31300::CUSTOMER_ACCESS_REG, als31300::CUSTOMER_ACCESS_REG_code)) {
-                LOG_ERROR("magneto: CANNOT INIT SLIDER SENSOR");
-                return kStatus_Fail;
-            }
+    std::int32_t init(xQueueHandle qHandle)
+    {
+        i2c = DriverI2C::Create(
+            static_cast<I2CInstances>(BoardDefinitions::MAGNETOMETER_I2C),
+            DriverI2CParams{.baudrate = static_cast<std::uint32_t>(BoardDefinitions::MAGNETOMETER_I2C_BAUDRATE)});
 
-            // CONFIGURATION register read
-            als31300::whole_reg_t read_reg;
-            i2cRead(als31300::CONF_REG, read_reg);
-            const als31300::conf_reg current_reg_conf(read_reg);
-            LOG_DEBUG("CONF read:\t%" PRIu32, static_cast<uint32_t>(current_reg_conf));
-            als31300::conf_reg reg_conf = current_reg_conf;
-            reg_conf.I2C_threshold      = als31300::CONF_REG_I2C_THRES_1v8;
-            reg_conf.int_latch_enable   = als31300::CONF_REG_LATCH_disabled; // we want to detect stable positions
-            reg_conf.channel_X_en       = als31300::CONF_REG_CHANNEL_enabled;
-            reg_conf.channel_Y_en       = als31300::CONF_REG_CHANNEL_enabled;
-            reg_conf.channel_Z_en       = als31300::CONF_REG_CHANNEL_disabled;
-            reg_conf.bandwidth          = static_cast<uint8_t>(BANDWIDTH_SELECT::bandwidth_7kHz);
-            if (current_reg_conf != reg_conf) {
-                [[maybe_unused]] auto ret = i2cWrite(als31300::CONF_REG, reg_conf);
-                assert(ret);
-                LOG_DEBUG("CONF wrote:\t%" PRIu32, static_cast<uint32_t>(reg_conf));
-
-                i2cRead(als31300::CONF_REG, read_reg);
-                LOG_DEBUG("CONF verify:\t%" PRIu32, static_cast<uint32_t>(als31300::conf_reg(read_reg)));
+        qHandleIrq = qHandle;
+
+        // Any configuration must be proceeded in active state
+        setActive(als31300::PWR_REG_SLEEP_MODE::active);
+
+        // Get write access
+        if (!i2cWrite(als31300::CUSTOMER_ACCESS_REG, als31300::CUSTOMER_ACCESS_REG_code)) {
+            LOG_ERROR("Failed to init magnetometer!");
+            return kStatus_Fail;
+        }
+
+        // Buffer for registers reading
+        als31300::whole_reg_t readRegister{};
+
+        // CONFIGURATION register read
+        auto readStatus                              = i2cRead(als31300::CONF_REG, readRegister);
+        const als31300::conf_reg currentConfRegister = readRegister;
+        if (readStatus) {
+            LOG_DEBUG("Current CONF register value: %" PRIu32, static_cast<std::uint32_t>(currentConfRegister));
+        }
+        else {
+            LOG_ERROR("Failed to read CONF register!");
```

Comment:
Shouldn't this mean that the whole function has failed and must return?

Reply:
As for the readouts - I don't think so, they're there only to not reconfigure the magnetometer if it's already configured properly. As for the writes - you're right, I'll add returns in case of config registers write failures